### PR TITLE
Add stale PRs management

### PR DESCRIPTION
**Pull request by jbonofre**

### CODE REVIEW

File: .github/workflows/stale.yml
Comment (nastra) [inline .github/workflows/stale.yml:55]: my gut feeling tells me this should probably be a bit higher (something around 90+ days)

File: .github/workflows/stale.yml
Comment (nastra) [inline .github/workflows/stale.yml:52]: Spark for example uses the following message: `We're closing this PR because it hasn't been updated in a while. This isn't a judgement on the merit of the PR in any way. It's just a way of keeping the PR queue manageable.
If you'd like to revive this PR, please reopen it and ask a committer to remove the Stale tag!`

I think we might also want to add some wording that makes it clear that the PR is marked as stale not because of the quality of the PR

Comment (jbonofre) [reply .github/workflows/stale.yml:52]: Good idea. Let me update the comment. Thanks !

**Comment by manuzhang** [comment]
@jbonofre Can you also increase `operations-per-run` to maybe 1000 or higher? With the current `100`, we are still processing stale issues in April, 2021, and it will become even slower when checking stale PRs as well.

File: .github/workflows/stale.yml
Comment (manuzhang) [reply .github/workflows/stale.yml:55]: I think 30 days is reasonable as a good reminder for author and reviewers.

**Comment by jbonofre** [comment]
@manuzhang sure ! Let me update accordingly

File: .github/workflows/stale.yml
Comment (Fokko) [inline .github/workflows/stale.yml:58]: Looks like they suggest numbers as high of 1000 for debugging: https://github.com/actions/stale?tab=readme-ov-file#operations-per-run

I believe 100 is already quite substantial and will generate a lot of noise. Raising this to 1k might flood people's mailboxes. WDYT?

Comment (jbonofre) [reply .github/workflows/stale.yml:58]: I set 1000 here for debugging first (to be back on 100 later). I think 100 is decent number (it's the same of the batch, it won't "limit" the issues/PRs considered to 100 actually). Let me back to 100 (debugging is probably not required if it works fine :)).

Comment (Fokko) [reply .github/workflows/stale.yml]: I saw @manuzhang's comment after I went over this. Maybe we should do some manual backlog cleaning. For example, close PRs of Spark/Flink versions that are not supported anymore.

**Review by Fokko — APPROVED** [review]
awesome

File: .github/workflows/stale.yml
Comment (jbonofre) [reply .github/workflows/stale.yml]: My original intention was to do a first run with 1000 and then back to 100 to do a first pass on PR that should be closed. But we can also do this first pass "manually".

File: .github/workflows/stale.yml
Comment (ajantha-bhat) [inline .github/workflows/stale.yml:54]: duplicate of line 50?

**Review by ajantha-bhat — APPROVED** [review]

**Review by nastra — APPROVED** [review]
overall this LGTM, but I'm a bit unsure about the 30 days. If pushing to the branch or commenting on the PR automatically counts as `activity`, then 30 days are probably ok

**Review by dimas-b — APPROVED** [review]

**Comment by nastra** [comment]
> > overall this LGTM, but I'm a bit unsure about the 30 days. If pushing to the branch or commenting on the PR automatically counts as `activity`, then 30 days are probably ok
> 
> Yes, as soon we have a new comment in a PR, the "counter" restart from zero. It means that we will have the stale flag only if we don't have any comment/activity on the PR since more than 30 days.

The commenting was clear to me. Does the same apply when you push to a branch but don't comment on it?

**Comment by manuzhang** [comment]
> Does the same apply when you push to a branch but don't comment on it?

Yes, as per [doc](https://github.com/actions/stale?tab=readme-ov-file#days-before-stale)
> The issues or the pull requests will be marked as stale if the last update (based on GitHub issue field updated_at) is older than the idle number of days.
It means that any updates made, or any comments added to the issues or to the pull requests will restart the counter of days before marking as stale.

@jbonofre Are you still working on this? A gentle reminder not to forget updating `operations-per-run`.

**Comment by jbonofre** [comment]
Yup. I reverted my change due to Fokko's comment. But I can change it back and adapt once merged.

**Comment by jbonofre** [comment]
@nastra @Fokko are we good to merge this one ?

**Review by nastra — APPROVED** [review]

**Comment by Fokko** [comment]
@jbonofre Yes, thanks everyone for chiming in for the review 🎉

**Comment by ajantha-bhat** [comment]
Looks like this never worked. This should fix it: https://github.com/apache/iceberg/pull/10706

**Pull request merged**
